Gems/Terrain/Code/Source/TerrainRenderer/ClipmapBounds.h

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#pragma once
+
+#include <AzCore/base.h>
+#include <AzCore/Math/Aabb.h>
+#include <AzCore/Math/Vector2.h>
+#include <AzCore/std/containers/vector.h>
+#include <TerrainRenderer/Vector2i.h>
+#include <TerrainRenderer/Aabb2i.h>
+
+namespace Terrain
+{
+
+    struct ClipmapBoundsDescritor
+    {
+        //! Size of the clipmap in width / height.
+        uint32_t m_size = 1024;
+
+        //! Current center location of the clipmap
+        AZ::Vector2 m_center = AZ::Vector2::CreateZero();
+
+        //! Unused area around edges of the clipmap. Center point must be moved by
+        //! at least this much before an update is triggered. This allows for larger
+        //! but less frequent updates.
+        uint32_t m_margin = 4;
+
+        //! Scale of the clip map compared to the world. A scale of 0.5 means that
+        //! a clipmap of size 1024 would cover 512 meters.
+        float m_scale = 1.0f;
+    };
+
+    struct ClipmapBoundsRegion
+    {
+        //! The world bounds of the updated region. Z is ignored.
+        AZ::Aabb m_worldAabb;
+
+        //! The clipmaps bounds of the updated region. Will always be between 0 and size.
+        //! Min inclusive, max exclusive. i.e., a single pixel would be min [0, 0], max [1, 1].
```

Comment:
Curious what @mbalfour-amzn thinks about this. I've only seen this way of describing inclusive / exclusive bounds recently, so it seems like this could be confusing for a lot of people reading the code that don't know what the brackets vs parenthesis mean. That's why I went super-explicit and even gave an example.

Reply:
Not a big deal either way, the description and example make sense, the notation just threw me a little because the comment explicitly states 'max exclusive' but square brackets indicate inclusivity.

Reply:
I'll remove the brackets entirely so as not to confuse with interval notation

Reply:
Because the square brackets have a well-defined mathematical meaning ( https://courses.lumenlearning.com/suny-osalgebratrig/chapter/domain-and-range/#:~:text=Parentheses%2C%20(%20or%20)%2C%20are,value%20is%20included%2C%20called%20inclusive. ), IMO it's far worse to use them _incorrectly_ and teach the wrong meaning to people who are unfamiliar, than to use them correctly and possibly confuse people who don't know the syntax.  

I'm fine with either "min [0, 0], max (1, 1)" or "x [0, 1), y [0, 1)".

Reply:
I included the example as an attempt to be more clear, but this is just causing confusion. I'll just keep it to "Min inclusive, max exclusive" with no example given.